java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -2327,24 +2327,106 @@ public final ColumnVector stringConcatenateListElements(Scalar separator,
+   * The caller is responsible for checking the output column size will not exceed the maximum size
+   * of a strings column (number of total characters is less than the value {@link Integer#MAX_VALUE}).
+   *
+   * @param repeatTimes The number of times each input string is repeated.
    * @return A new java column vector containing repeated strings.
    */
   public final ColumnVector repeatStrings(int repeatTimes) {
-    assert type.equals(DType.STRING) : "column type must be a String";
-
+    assert type.equals(DType.STRING) : "column type must be String";
     return new ColumnVector(repeatStrings(getNativeView(), repeatTimes));
   }
 
+  /**
+   * Given a strings column, an output strings column is generated by repeating each of the input
+   * string by a number of times given by the corresponding row in a <code>repeatTimes</code>
+   * numeric column.
+   *
+   * In special cases:
+   *  - Any null row (from either the input strings column or the <code>repeatTimes</code> column)
+   *    will always result in a null output string.
+   *  - If any value in the <code>repeatTimes</code> column is not a positive number and its
+   *    corresponding input string is not null, the output string will be an empty string.
+   *
+   * The caller is responsible for checking the output column size will not exceed the maximum size
+   * of a strings column (number of total characters is less than the value {@link Integer#MAX_VALUE}).
+   *
+   * @param repeatTimes The column containing numbers of times each input string is repeated.
+   * @return A new java column vector containing repeated strings.
+   */
+  public final ColumnVector repeatStrings(ColumnView repeatTimes) {
+    assert type.equals(DType.STRING) : "column type must be String";
+    return new ColumnVector(repeatStringsWithColumnRepeatTimes(getNativeView(),
+            repeatTimes.getNativeView(), 0));
+  }
+
+  /**
+   * This function is an overloaded version of {@link #repeatStrings(ColumnView) repeatStrings},
+   * with an additional parameter <code>outputStringSizes</code> that provides a column containing
+   * the pre-computed sizes of the output strings.
+   *
+   * @param repeatTimes The column containing numbers of times each input string is repeated.
+   * @param outputStringSizes A numeric column providing the pre-computed sizes of the output strings.
+   * @return A new java column vector containing repeated strings.
+   */
+  public final ColumnVector repeatStrings(ColumnView repeatTimes, ColumnView outputStringSizes) {
+    assert type.equals(DType.STRING) : "column type must be String";
+    return new ColumnVector(repeatStringsWithColumnRepeatTimes(getNativeView(),
+            repeatTimes.getNativeView(), outputStringSizes.getNativeView()));
+  }
+
+  /**
+   * Struct to hold the results output from the function
+   * {@link #repeatStringsSizes(ColumnView) repeatStringsSizes}.
+   */
+  static final class StringSizes implements AutoCloseable {
```

Comment:
This class is returned from a public interface and therefore must be public to be accessed.
```suggestion
  public static final class StringSizes implements AutoCloseable {
```